File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    cnt = 0
    
    for i in range(n, 0, -1):
        target = n
        j = i
        
        while target > 0:
            target -= j
            if (j := j - 1) <= 0:
                break
        
        if target == 0:
            cnt += 1

    return cnt

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
public class Solution {
    public static int solution(int n) {
        int cnt = 0;

        for (int i = n; i > 0; i--) {
            int target = n;
            int j = i;

            while (target > 0 && j > 0) {
                target -= j;
                j--;
            }

            if (target == 0) cnt++;
        }

        return cnt;
    }
}